tests/virt_autotest/validate_system_health.pm

```diff
@@ -0,0 +1,73 @@
+# SUSE's openQA tests
+#
+# Copyright 2012-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Do a basic examination to the host and guests after tests run
+# Maintainer: Julie CAO <JCao@suse.com>, qe-virt@suse.de
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use ipmi_backend_utils qw(reconnect_when_ssh_console_broken);
+use Utils::Architectures;
+use virt_autotest::common;
+use virt_autotest::utils qw(start_guests check_host_health check_guest_health);
+use virt_utils qw(collect_host_and_guest_logs);
+
+sub prepare_run_test {
+    reconnect_when_ssh_console_broken unless defined(script_run("rm -f /root/{commands_history,commands_failure}", die_on_timeout => 0));
+    script_run("history -c");
+}
+
+sub run_test {
+
+    my $self = shift;
+    my %health_status = ();
+    @health_status{(keys %virt_autotest::common::guests)} = ();
+    $health_status{host} = check_host_health;
+    foreach my $guest (grep { $_ ne 'host' } keys %health_status) {
+        script_run("virsh start $guest", die_on_timeout => 0);
+        if (script_retry("nmap $guest -PN -p ssh | grep open", delay => 2, retry => 60, die => 0) == 0) {
+            $health_status{$guest} = check_guest_health($guest);
+        }
+        else {
+            $health_status{$guest} = "Unreachable";
+        }
+        record_info("Failed guest", "$guest: $health_status{$guest}", result => 'fail') unless $health_status{$guest} eq 'pass';
+    }
+    my @unhealthy_list = grep { $health_status{$_} ne 'pass' } keys %health_status;
+
+    # Upload logs on x86_64
+    if (is_x86_64 && (get_var('FORCE_UPLOAD_LOGS') || @unhealthy_list != 0)) {
+        if (get_var('FORCE_UPLOAD_LOGS')) {
+            collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } keys %health_status));
+        }
+        else {
+            collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } @unhealthy_list));
+        }
+        $self->upload_coredumps;
+        save_screenshot;
+    }
+    die "Host or guests are not healthy!" unless @unhealthy_list == 0;
+}
+
+sub post_fail_hook {
+    my $self = shift;
+    diag("Module validate_system_health post fail hook starts.");
+    $self->junit_log_provision((caller(0))[3]);
+    reconnect_when_ssh_console_broken unless defined(script_run("history -w /root/commands_history", die_on_timeout => 0));
+    $self->upload_coredumps;
```

Comment:
I think you can also remove ```upload_coredumps``` here.

Reply:
Are coredump files uploaded by supportconfig? I did not see them.

Reply:
You already uploaded it in 
```
if (is_x86_64 && (get_var('FORCE_UPLOAD_LOGS') || @unhealthy_list != 0)) {
```

Reply:
oh, they are different. if collect_logs fails in test_run, coredump files are still needed in post_fail_hook() since there may be coredumps.